src/containers/setContacts/index.tsx

```diff
@@ -0,0 +1,202 @@
+import { Alert, Spin } from 'antd';
+import moment from 'moment';
+import React, { useEffect } from 'react';
+import { connect, useDispatch } from 'react-redux';
+import { useHistory, useLocation } from 'react-router';
+import protectedApiClient from '../../api/protectedApiClient';
+import { Routes } from '../../App';
+import ContactsForm from '../../components/ContactsForm';
+import { C4CState } from '../../store';
+import {
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+} from '../../utils/asyncRequest';
+import { encodeProfileFieldFile } from '../../utils/fileEncoding';
+import {
+  participatingFamilySearchQuery,
+  participatingFamilySearchQueryFlag,
+} from '../../utils/signupFlow';
+import {
+  getContactInfo,
+  resetSetContactState,
+  setContactInfo,
+} from './ducks/thunks';
+import {
+  AdditionalContact,
+  Child,
+  ContactFormFields,
+  ContactInfo,
+  ContactsReducerState,
+} from './ducks/types';
+
+interface SetContactsProps {
+  readonly contacts: ContactsReducerState['contacts'];
+  readonly setContacts: ContactsReducerState['setContacts'];
+}
+
+const SetContacts: React.FC<SetContactsProps> = ({ contacts, setContacts }) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+  useEffect(() => {
+    dispatch(getContactInfo());
+  }, [dispatch]);
+  const registeringAsParticipatingFamily =
+    new URLSearchParams(location.search).get(
+      participatingFamilySearchQueryFlag,
+    ) != null;
+
+  if (asyncRequestIsComplete(setContacts) && registeringAsParticipatingFamily) {
+    protectedApiClient.makePFRequest();
+    history.push({
+      pathname: Routes.SIGNUP_CONFIRMATION,
+      search: participatingFamilySearchQuery,
+    });
+  }
+
+  function selectProfilePictureOrDefaultToUndefined(
+    i: number,
+    arr: Child[] | AdditionalContact[],
+  ): string | undefined {
+    if (asyncRequestIsComplete(contacts) && i < arr.length) {
+      return arr[i].profilePicture;
+    } else {
+      return undefined;
+    }
+  }
+
+  const handleCloseSuccessAlert = () => {
+    dispatch(resetSetContactState());
+  };
+
+  const onFinish = async (values: ContactFormFields) => {
+    if (!asyncRequestIsComplete(contacts)) {
+      throw new Error(
+        'Setting Form values requires knowing current contact information',
+      );
+    }
+
+    dispatch(
+      setContactInfo({
+        mainContact: {
+          email: contacts.result.mainContact.email,
+          firstName: values.firstName,
```

Comment:
Maybe deconstruct `values` and pull out all the properties that you don't need, like:
`const {notNeeded1, notNeeded2, notNeeded3, ...requiredValues} = values;`

so then, you could do:
```
mainContact: {
   ...requiredValues,
   ...
}
```

I'm not sure how much this would help because for some of them properties of `values` you do conditionals, like for for pronouns, allergies, and diagnosis.